mlflow/sagemaker/cli.py

```diff
@@ -358,22 +358,26 @@ def build_and_push_container(build, push, container, env_manager, mlflow_home):
         click.echo("skipping both build and push, have nothing to do!")
     if build:
         sagemaker_image_entrypoint = (
-            'ENTRYPOINT ["python", "-c", "import sys; from mlflow.models import container as C; '
-            f"C._init(sys.argv[1], '{env_manager}')\"]"
+            "import sys; from mlflow.models import container as C; "
+            f"C._init(sys.argv[1], '{env_manager}')"
```

Comment:
Unrelated to this PR, creating a module like below and running that using `python -m mlflow.foo.bar` might be cleaner and harder to make mistakes:

```
import sys
from mlflow.models import container

container._init(sys.argv[1], sys.argv[2])
```

Reply:
Yup we may want to do that when the script becomes more complex. Let me keep it as it is in the scope of this PR.